esp/services/ws_dali/ws_daliservice.cpp

```diff
@@ -23,17 +23,160 @@
+        if (!child)
+            throw makeStringExceptionV(ECLWATCH_INTERNAL_ERROR, "Couldn't find %s/%s", head.str(), tail);
+
+        if (!req.getNoBackup())
+            backupSDSData(child, path, tail);
+
+        root->removeTree(child);
+        child.clear();
+        root.clear();
+        conn->commit();
+        conn->close();
+        resp.setMessage("Dali updated.");
+    }
+    catch(IException* e)
+    {
+        FORWARDEXCEPTION(context, e,  ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+void CWSDaliEx::importSDSData(const char* dataXML, const char* dataFile, const char* path, bool add)
+{
+    StringBuffer head, tmp;
+    const char* tail = splitpath(path, head, tmp);
+    if (!add)
+    {
+        Owned<IRemoteConnection> bconn = querySDS().connect(remLeading(path), myProcessSession(), RTM_LOCK_READ|RTM_SUB, daliConnectTimeoutMs);
+        if (bconn)
+        {
+            Owned<IPropertyTree> broot = bconn->getRoot();
+            backupSDSData(broot, path, tail);
+        }
+    }
+
+    Owned<IRemoteConnection> conn = querySDS().connect(head, myProcessSession(), 0, daliConnectTimeoutMs);
+    if (!conn)
+        throw makeStringExceptionV(ECLWATCH_CANNOT_CONNECT_DALI, "Could not connect to %s", path);
+
+    StringAttr newtail; // must be declared outside the following if
+    Owned<IPropertyTree> root = conn->getRoot();
+    if (!add)
+    {
+        Owned<IPropertyTree> child = root->getPropTree(tail);
+        root->removeTree(child);
+
+        //If replacing a qualified branch then remove the qualifiers before calling addProp
+        const char* qualifier = strchr(tail, '[');
+        if (qualifier)
+        {
+            newtail.set(tail, qualifier-tail);
+            tail = newtail;
+        }
+    }
+
+    Owned<IPropertyTree> branch;
+    if (!isEmptyString(dataXML))
+        branch.setown(createPTreeFromXMLString(dataXML));
+    else
+    {
+        branch.setown(createPTreeFromXMLFile(dataFile));
+        removeFileTraceIfFail(dataFile);
+    }
+
+    Owned<IPropertyTree> oldEnvironment;
+    if (streq(path, "Environment")) //This is from daliadmin code. Should also check the path eq. '/Environment' (see the 7th line below)?
+        oldEnvironment.setown(createPTreeFromIPT(conn->queryRoot()));
+    root->addPropTree(tail, LINK(branch));
+    conn->commit();
+    PROGLOG("Branch %s imported", path);
+    conn->close();
+
+    if (*path=='/')
+        path++;
+    if (streq(path, "Environment"))
+     {
+        PROGLOG("Refreshing cluster groups from Environment");
+        StringBuffer response;
+        initClusterGroups(false, response, oldEnvironment);
+        if (!response.isEmpty())
+            PROGLOG("updating Environment via import path=%s : %s", path, response.str());
+    }
+}
+
+void CWSDaliEx::backupSDSData(IPropertyTree* srcPTree, const char* srcPath, const char* prefix)
+{
+    StringBuffer bakName;
+    Owned<IFileIO> io = createUniqueFile(daliFolder, prefix, "bak", bakName);
```

Comment:
"daliFolder" is not really _the_ dali folder (that Dali uses), so perhaps call this espDaliBackupFolder or similar.

How would this work in the cloud/containerized setup?
Writing locally would be lost once pod goes away.

Reply:
How about renaming the "daliFolder" to espDaliDataFolder? It is used for both backup and temp export folder.
Should I use the storage planes for cloud? Any example?